Bks.DataAccess.Mongo.Test/MongoConnectorTest.cs

```diff
@@ -0,0 +1,69 @@
+using System;
+using System.Collections.Generic;
+using FakeItEasy;
+using FluentAssertions;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using MongoDB.Driver;
+
+namespace Bks.DataAccess.Mongo.Test
+{
+    [TestClass]
+    public class MongoConnectorTest
+    {
+        private const string DbName = "MY_DB_NAME";
+        private const string DbPrefix = "MY_DB_PREFIX";
+        private const string DbCollectionName = "MY_COLLECTION_NAME";
+
+        private readonly ILogger<MongoConnector> mockLogger = 
+            A.Fake<ILogger<MongoConnector>>();
+        
+        private readonly IReadOnlyCollection<IMongoClassMapper> classMaps =
+            A.Fake<IReadOnlyCollection<IMongoClassMapper>>();
+
+        private static IOptions<MongoSettings> CreateMongoSettingsOptions()
+        {
+            var mongoSettings = new MongoSettings()
+            {
+                ConnectionString = "mongodb://localhost:27017?maxPoolSize=5",
+                Database = DbName,
+                CollectionPrefix = DbPrefix
+            };
+            var mongoSettingsOptions = Options.Create(mongoSettings);
+            return mongoSettingsOptions;
+        }
+
+        [TestMethod]
+        public void RegisterClassMaps_with3Mappers_allMappersAreUsed()
+        {
+            var mapperA = A.Fake<IMongoClassMapper>();
+            var mapperB = A.Fake<IMongoClassMapper>();
+            var mapperC = A.Fake<IMongoClassMapper>();
+            IReadOnlyCollection<IMongoClassMapper> mockedClassMaps = new List<IMongoClassMapper>(){ mapperA , mapperB, mapperC };
+
+            var options = CreateMongoSettingsOptions();
+            var mockMongo = A.Fake<MongoConnector>(
+                x => x.WithArgumentsForConstructor(
+                    new object[] { options, mockLogger, mockedClassMaps }));
+            A.CallTo(() => mapperA.Map()).MustHaveHappened();
+            A.CallTo(() => mapperB.Map()).MustHaveHappened();
+            A.CallTo(() => mapperC.Map()).MustHaveHappened();
+        }
+
+        [TestMethod]
+        public void GetCollection_withPrefix_ShouldReturnAnInstanceAccordingToNamingConventions()
```

Comment:
Rename to GetCollection()
The meaning of that test is that you can get collection :).

I don't understand what are you testing here (and again you use mock of mongo). The risht test should define an expected result for call to database.GetCollection and check the result

Reply:
Testing the The mongoConnection creation of collection according to the naming rules.
A name Should be generated according to a standard of :
$"{DbName}.{DbPrefix}_{DbCollectionName}"